app/controllers/posts_controller.rb

```diff
@@ -28,6 +28,16 @@ def create
     end
   end
 
+  def destroy
+    post = Post.find(params[:id])
+    user = User.find(post.user_id)
+    user.posts_counter -= 1
+    post.destroy
+    user.save
+    flash[:success] = 'You have deleted this post successfully!'
+    redirect_to user_path(current_user.id)
+  end
+
   private
```

Comment:
- [ ] Kindly add `load_and_authorize_resource` to this posts_controller so that CanCanCan could be able to authorize users based on the rules you set in your CanCanCan `ability` class. Kindly go through this [article](https://github.com/CanCanCommunity/cancancan)

Reply:
Thank you, added `load_and_authorize_resource` to controllers